components/shared_code/src/shared_data_model/sources/bitbucket.py

```diff
@@ -0,0 +1,78 @@
+"""Bitbucket source."""
+
+from pydantic import HttpUrl
+
+from shared_data_model.meta.entity import Color, Entity, EntityAttribute, EntityAttributeType
+from shared_data_model.meta.source import Source
+from shared_data_model.parameters import (
+    URL,
+    Branch,
+    Branches,
+    BranchesToIgnore,
+    BranchMergeStatus,
+    Days,
+    FailureType,
+    MergeRequestState,
+    MultipleChoiceParameter,
+    MultipleChoiceWithAdditionParameter,
+    PrivateToken,
+    ResultType,
+    StringParameter,
+    TargetBranchesToInclude,
+)
+
+BITBUCKET_BRANCH_HELP_URL = HttpUrl("https://confluence.atlassian.com/bitbucketserver/branches-776639968.html")
+
+BITBUCKET = Source(
+    name="Bitbucket",
+    description="Bitbucket Cloud is a Git based code hosting and collaboration tool, built for teams."
+                "Bitbucket's best-in-class Jira and Trello integrations are designed to bring the entire software team together to execute on a project.",
```

Comment:
Maybe leave out that second sentence, it sounds more like a commercial than a factual description.

Reply:
Also, tests are failing because you didn't add a bitbucket logo yet.